### ISSUE TITLE
CategoricalDomain new category when predict 

### ISSUE THREAD

**Issue by siriJR** [jpmml/sklearn2pmml#203]
Hello, author.
When I use the model to predict，it reports an error:
“sklean2pmml ['col_tm']: Data contains 1 invalid values”

This column does not contain all enumeration values in the training data set。

How can I solve this problem, thank you~

**Comment by vruusmann** [comment]
TLDR: You are asking your model to make a prediction about something that the model knows nothing about. Is it a good idea or not?

> How can I solve this problem

Your model is outdated - the data has changed in a way so that it is now outside of the scope of the original model. I would advise you to re-train the model.

Is this new category level "very similar" to some existing category level? If it is, then you can replace the new value with this old value:
```python
CategoricalDomain(invalid_value_treatment = "as_is", invalid_value_replacement = "default_category_value")
```

If this new category level is completely novel information, then you should replace it with the missing value, and use some imputation strategy (ie. replace it with the median category value):
```python
[CategoricalDomain(invalid_value_treatment = "as_missing"), SimpleImputer(strategy = "median")]
```
